crates/resolver/src/config.rs

```diff
@@ -443,7 +443,7 @@ impl ProtocolConfig {
 #[cfg_attr(
     feature = "serde",
     derive(Serialize, Deserialize),
-    serde(default, deny_unknown_fields)
```

Comment:
I had to remove `deny_unknown_fields` here, or `config_tests::test_read_config` finds `timeout` as an unknown field, even though it comes from the flattened `ConnectionOptions` (or `NameServerOptions.ConnectionOptions.timeout` in subsequent commits).

Removing `deny_unknown_fields` also required a `#[ignore]` on the `test_reject_unknown_fields()` test. If we decide to go without `deny_unknown_fields`, I'll delete it. Otherwise I'll have to un-flatten a layer in the TOML I think.

Reply:
I don't think we want to get rid of `deny_unknown_fields`, it seems important to me.